### PR TITLE
Add example data loading

### PR DESCRIPTION
**Pull request by sethaxen**
This PR replaces our function `load_arviz_data`, which until now forwarded to Python, with a Julia function `load_example_data`, which similarly loads example `InferenceData` objects from local and remote netCDF files. `load_arviz_data` is now deprecated.

Like Python ArviZ, we distribute 2 objects with 8-schools data in the repo in `deps/data/`, which can be used even if the user is offline, and remote datasets are also defined. All data management is handled by DataDeps.jl.

Internally, `load_example_data` calls `from_netcdf`, which still dispatches to Python, so this solution is not yet pure Julia.

### CODE REVIEW

**Comment by sethaxen** [comment]
Much of the example data code is directly taken from Python ArviZ. I wonder if it would make sense to have an arviz_example_data repo that uses JSON or something to define the datasets. Then both packages could reference that repo.

We already have such a repo https://github.com/arviz-devs/arviz_example_data, but AFAICT it just contains extra information or files used to generate some of the data.

Thoughts @OriolAbril @ahartikainen ?

**Comment by ahartikainen** [comment]
This is something we could do.

File: src/example_data.jl [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+function _register_data(md::LocalFileMetadata)
+    return DataDeps.register(DataDeps.ManualDataDep(md.name, md.description))
+end
+
+function _register_data(md::RemoteFileMetadata)
+    return DataDeps.register(DataDeps.DataDep(md.name, md.description, md.url, md.checksum))
+end
+
+_init_data_deps() = map(_register_data, values(EXAMPLE_DATA))
```
Comment (sethaxen) [inline src/example_data.jl:229]: @oxinabox, registering the 9 datasets in `EXAMPLE_DATA` seems to increase package load time by 300ms. That seems high. Am I using DataDeps incorrectly somehow?

**Comment by sethaxen** [comment]
Since DataDeps is an internal only implementation detail, for now we can accept the extra 300ms load time, but it may be worth it  exploring other solutions or rolling our own to do everything at compile time instead of at init time. We can do this at a later time.

**Pull request merged**